### ISSUE TITLE
Algorithm implementation of polyfill

### ISSUE THREAD

**Issue by yangscout** [uber/h3-py#300]
I am very interested in the polyfill function when using h3. I want to find the specific definition of this function, but I can't find it in the local library or github. Can you share the specific code implementation?

**Comment by kylebarron** [comment]
I _think_ the existing polyfill implementation checks which cell _centers_ intersects the geometry of interest. It does not check against actual cell boundaries

**Comment by dfellis** [comment]
That's correct. Using cell center containment guarantees that two contiguous polygons (with perfect overlap of some of the points of the polygon, like ZCTAs or other regional boundaries) will produce perfectly contiguous sets of H3 cells where there are zero cells that end up in both sets.

That was a hard requirement for the conversion of Uber's internal geofences to H3 during a transition phase where some surge algorithms continued operating on distinct geofences but needed to be rendered in conjunction with other algorithms that used H3 natively.

One caveat, though, is that cell center containment is done in `lat, lng` coordinates, so it's effectively operating in mercator projection while most of H3 is using an icosahedron-centered gnomonic projection, so at very large scales there are discrepancies there. However, since all Uber geofences were defined in mercator projection, and that is also the de facto standard for GeoJSON and etc, it's probably the best choice, despite feeling a bit off compared to the rest of the library.

Now that H3 4.0 has made it possible to start adding new polyfill algorithms, though, I'm hoping to make other containment and projection combinations available. :) (Once I have some spare time, that is. :/ )

**Comment by yangscout** [comment]
Yes, the current polyfill does not meet my needs perfectly. I want to ask where I can see the algorithm implemented by the polyfill and improve it.

**Comment by dfellis** [comment]
> Yes, the current polyfill does not meet my needs perfectly. I want to ask where I can see the algorithm implemented by the polyfill and improve it.

So the implementation [lives here](https://github.com/uber/h3/blob/master/src/h3lib/lib/algos.c#L891) but it hasn't been refactored to use the `flags` argument, yet, as a warning for you. What particular polyfill algorithm do you need?

**Comment by yangscout** [comment]
> > Yes, the current polyfill does not meet my needs perfectly. I want to ask where I can see the algorithm implemented by the polyfill and improve it.
> 
> So the implementation [lives here](https://github.com/uber/h3/blob/master/src/h3lib/lib/algos.c#L891) but it hasn't been refactored to use the `flags` argument, yet, as a warning for you. What particular polyfill algorithm do you need?

The current polyfill algorithm is to find all h3 points whose mesh center is located in the polygon. I want to find an algorithm to find that all the vertices of its h3 hexagon are located in the polygon
